x-pack/plugins/apm/server/routes/index_pattern.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { InternalCoreSetup } from 'src/core/server';
+import { getIndexPattern } from '../lib/index_pattern';
+
+const ROOT = '/api/apm/index_pattern';
+const defaultErrorHandler = (err: Error) => {
+  // eslint-disable-next-line
+  console.error(err.stack);
+  throw Boom.boomify(err, { statusCode: 400 });
```

Comment:
Minor nit: is there any benefit to making this more informative by providing different status codes? For example, if the error thrown is due to a JS error (e.g. accessing a property on `undefined`) then this would be a 500.

Reply:
> This code looks great, I just had some questions and suggestions. I would approve the PR but when I tested locally I noticed that the problem in the referenced issue still occurs:
> 
> ![image](https://user-images.githubusercontent.com/1238659/58897334-e8382000-86ac-11e9-8ec9-53a673ad18f4.png)
> 
> So I think either this PR needs more work to fix the issue or we should update the PR description to clarify that it doesn't fix the reported problem.

In my testing, it has been working as expected. The intention is to create the APM index pattern automatically, when the user performs and action which requires it. In my testing, if i start kibana with an empty ES cluster, i see no errors like this until after the APM home page loads (which requires the apm index pattern for the kuery bar to work).